### PR TITLE
Dhp 228 disconnect device

### PR DESCRIPTION
**Pull request by reblg**
## Description of change
This PR allows veterans to opt to disconnect their fitbit device so that it will stop sharing data with the VA. It simply flips the `active` boolean is a Veteran Device Record from true to false. 

## Tests
- Unit testing

## Things to know about this PR
- Are there additions to a settings.yml file? Do they vary by environment?
- Pre-existing feature flag: dhp_connected_devices_fitbit
- Is there some Sentry logging that was added? What alerts are relevant?
- Are there any Prometheus metrics being collected? What Grafana dashboard were they added do?
- Are there Swagger docs that were updated? Yes dhp swagger docs were updated for `fitbit/disconnect` endpoint
- Is there any PII concerns or questions? 
- Modified existing tests for this issue

### CODE REVIEW

**Review by jhwuz — APPROVED** [review]

File: modules/dhp_connected_devices/spec/services/dhp_connected_devices/veteran_device_records_service_spec.rb [hunk 1]
```diff
@@ -57,4 +57,39 @@
...
+
+    context 'when veteran device record, device exists, and active is set to true' do
+      it 'updates record' do
+        VeteranDeviceRecordsService.deactivate(user, device.key)
+        record = VeteranDeviceRecord.all.first
+        expect(record.active).to be(false)
+      end
+    end
+
+    context 'when veteran device record doesn\'t exist' do
```
Comment (yonidoronpeters) [inline modules/dhp_connected_devices/spec/services/dhp_connected_devices/veteran_device_records_service_spec.rb:77]:
```suggestion
    context 'when device doesn\'t exist' do
```

**Review by yonidoronpeters — APPROVED** [review]
Awesome! a couple of improvement suggestions ⭐

File: modules/dhp_connected_devices/spec/models/device_spec.rb [hunk 2]
```diff
@@ -5,6 +5,19 @@
...
+
+  it 'has a unique device name' do
+    Device.create(name: 'fitbit')
+    expect(Device.create(name: 'fitbit')).not_to be_valid
+  end
+
+  it 'cannot create a device without a name' do
+    expect(Device.new).not_to be_valid
+  end
+
```
[the remark this answers is not in the document]
Comment (reblg) [reply modules/dhp_connected_devices/spec/models/device_spec.rb:20]: thank you!

File: modules/dhp_connected_devices/spec/services/dhp_connected_devices/veteran_device_records_service_spec.rb [hunk 3]
```diff
@@ -57,4 +57,39 @@
...
+
+  describe 'veteran_device_records#deactivate' do
+    let!(:user) { create :user, :loa3 }
+    let!(:device) { create :device, :fitbit }
+
+    before do
+      @vdr = VeteranDeviceRecord.create(icn: user.icn, device_id: device.id, active: true)
+    end
+
+    context 'when veteran device record, device exists, and active is set to true' do
```
[the remark this answers is not in the document]
Comment (reblg) [reply modules/dhp_connected_devices/spec/services/dhp_connected_devices/veteran_device_records_service_spec.rb:69]: Will fix in a different commit.

File: modules/dhp_connected_devices/spec/services/dhp_connected_devices/veteran_device_records_service_spec.rb (hunk 1 above)
Comment (reblg) [reply modules/dhp_connected_devices/spec/services/dhp_connected_devices/veteran_device_records_service_spec.rb:77]: Will fix  in another commit

**Pull request merged**